components/codec/src/columnar/mod.rs

```diff
@@ -255,7 +262,7 @@ impl ColumnarEncoder {
         let enc = ValuesEncoderImpl::default();
         let data_size = match hint.datum_kind {
             DatumKind::Null => 0,
-            DatumKind::Timestamp => enc.estimated_encoded_size(
+            DatumKind::Timestamp => TimestampEncoderImpl.estimated_encoded_size(
```

Comment:
Shall use the `ValuesEncoderImpl` here? we can make `ValuesEncoderImpl` to delegate the encoding work with the `TimestampEncoderImpl` as the underlying encoder.